Sources/Cells/UserAdTableViewCell/UserAdTableViewCell.swift

```diff
@@ -0,0 +1,220 @@
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Private methods
+
+    private func setup() {
+        isAccessibilityElement = true
+        backgroundColor = .bgPrimary
+        selectionStyle = .none
+
+        contentView.addSubview(contentStack)
+        contentView.addSubview(ribbonView)
+
+        contentStack.addArrangedSubview(adImageView)
+        contentStack.addArrangedSubview(descriptionStack)
+
+        descriptionStack.addArrangedSubview(titleLabel)
+        descriptionStack.addArrangedSubview(subtitleLabel)
+        descriptionStack.addArrangedSubview(detailLabel)
+
+        descriptionStack.setCustomSpacing(.verySmallSpacing, after: titleLabel)
+        descriptionStack.setCustomSpacing(.mediumSpacing, after: subtitleLabel)
+
+        setupStyleConstraints()
+    }
+
+    private func setupStyleConstraints() {
+        NSLayoutConstraint.deactivate(compactConstraints + regularConstraints)
+
+        let constraints = style == .compressed ? compactConstraints : regularConstraints
+        NSLayoutConstraint.activate(sharedConstraints + constraints)
+    }
+
+    // MARK: Public methods
+
+    public func configure(with style: Style, model: UserAdTableViewCellViewModel) {
+        self.style = style
+        self.model = model
+
+        accessibilityLabel = model.accessibilityLabel
+
+        let imageInset = style == .compressed
+            ? UserAdTableViewCell.compressedImageWidth
+            : UserAdTableViewCell.defaultImageWidth
+
+        separatorInset = .leadingInset(.largeSpacing + imageInset)
+        contentStack.alignment = style == .compressed ? .center : .top
+
+        ribbonView.style = model.ribbon.style
+        ribbonView.title = model.ribbon.title
+
+        titleLabel.text = model.titleText
+        titleLabel.numberOfLines = style == .compressed ? 1 : 2
+
+        if let subtitle = model.subtitleText {
+            subtitleLabel.text = subtitle
+            subtitleLabel.isHidden = false
+        }
+
+        if let detail = model.detailText {
+            detailLabel.text = detail
+            detailLabel.isHidden = style == .compressed
+        }
+    }
+
+    // MARK: - Overrides
+
+    public override func prepareForReuse() {
+        adImageView.cancelLoading()
+        adImageView.image = nil
+
+        ribbonView.title = ""
+        ribbonView.style = .default
+
+        titleLabel.text = nil
+
+        [subtitleLabel, detailLabel].forEach {
+            $0.text = nil
+            $0.isHidden = true
+        }
+
+        super.prepareForReuse()
+    }
+
+}
+
+// MARK: - ImageLoading
+
+extension UserAdTableViewCell: ImageLoading {
+
+    public func loadImage() {
+        if let imagePath = model?.imagePath {
+            adImageView.loadImage(for: imagePath, imageWidth: .veryLargeSpacing, loadingColor: loadingColor, fallbackImage: fallbackImage)
```

Comment:
We should probaby try to fetch the image with the correct imageWidth here.

```suggestion
            let imageWidth = style == .compressed ? compressedImageWidth : defaultImageWidth
            adImageView.loadImage(for: imagePath, imageWidth: imageWidth, loadingColor: loadingColor, fallbackImage: fallbackImage)
```